rclcpp/src/rclcpp/qos_event.cpp

```diff
@@ -0,0 +1,49 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rclcpp/qos_event.hpp"
+
+
+namespace rclcpp
+{
+
+/// Get the number of ready events
+size_t
+QOSEventBase::get_number_of_ready_events()
+{
+  return 1;
```

Comment:
Add a `//TODO` or something here so we don't completely lose it.

Reply:
This function is actually done, no TODO here :)

Reply:
When you say it's done do you mean returning 1 is the final implementation or that it's done and not uploaded yet?

Reply:
For this class there will always be one event is what @mm318 is saying

Reply:
Maybe I'm misunderstanding what a "ready_event" is. I assumed that it was referring to the number of events that had occured/could be taken in the rmw layer. If so then I don't get how we can always assume this will be 1. Is this not what this is used for?

Reply:
I don't know why they decided to attach the "ready" qualifier to the name of the member functions of the `Waitable` interface. The return value of this function is actually supposed to be more static than what the name of the function would lead you to believe.

The meaning of the return of value of this function is actually just the number of `rcl_event_t` handles this `Waitable` class has.